src/sqlfluff/core/linter/runner.py

```diff
@@ -92,6 +93,8 @@ def run(self, fnames, fix):
         for fname, partial in self.iter_partials(fnames, fix=fix):
             try:
                 yield partial()
+            except (bdb.BdbQuit, KeyboardInterrupt):
```

Comment:
Unrelated to the overall issue, but this makes SQLFluff better behaved: If a user or developer hits ^C at runtime or quits out of the debugger, don't catch the exception.

Reply:
```suggestion
            except (bdb.BdbQuit, KeyboardInterrupt):  #pragma: no cover
```